### ISSUE TITLE
One screen in Dash's desktop tab sometimes freezes

### ISSUE THREAD

**Issue by JackTheFoxOtter** [Yellow-Dog-Man/Resonite-Issues#1063]
### Describe the bug?

I have a multi monitor setup. Sometimes, one of the views in the Dash's desktop tab freezes. With that I mean only one of the three monitors will stop rendering, the others will still work fine, the frozen tab shows a freeze frame from some point earlier, still proxies input to the OS (I can still click / scroll etc.), it's just the rendered view that isn't updating.

The only way to fix this is restarting.

I've never had this happen on more than one monitor view at the same time, and it's also not always the same one. I don't think I've ever noticed the middle monitor freeze, but that's also the one I use the least, so I might have not noticed.

This happens rare enough that I don't know how to reproduce it, but frequently enough that it's rather frustrating. (About one or twice per week when I play enough I'd say. Maybe it happens more frequently, but that's how often I notice.)

Also, I've never had this happen the first time I open the desktop tab. It's always after all screens worked for a while and I interacted with them.

### To Reproduce

If I knew I'd put them here...

### Expected behavior

None of the three screens on the Dash's desktop tab should freeze.

### Screenshots

Not much point in taking a screenshot of a frozen screen (also personal information on my desktop)
But here is my screen configuration on Windows, if it contains any valuable information:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/29d1fb74-09b9-4fbc-b101-f955de3e9f1b)

Main screen:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/0ff6f8d0-2049-4b64-b7be-6cd209a13685)

Left screen:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/0d4a20e8-4bfc-44cf-a1b3-ddffadb3a1f0)

Right screen:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/15dabe09-2ce7-47d1-a6ad-543eeec6b794)

### Resonite Version Number

2024.1.3.393

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Valve Index / SteamVR

### Log Files

[J4-C - 2024.1.3.393 - 2024-01-03 19_28_04.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13822707/J4-C.-.2024.1.3.393.-.2024-01-03.19_28_04.log)

I can't tell when it happened in this log, I shut the game down a couple of minutes after I noticed.

**Comment by JackTheFoxOtter** [comment]
As an additional note, this has happened over multiple NVidia driver versions and even multiple Windows installs, back on Windows 10 as well as now on Windows 11. I don't think it's a OS issue, but it might be something about my hardware configuration that's causing it, since that hasn't changed in a while.

**Comment by JackTheFoxOtter** [comment]
> Is this issue still relevant, @JackTheFoxOtter? Have you isolated any particular replication conditions, or found any other users with this issue? Does the issue occur if all three monitors are set to the same res and refresh rate?
> 
> Could you investigate the UIX when this occurs to see if anything is out of place/incorrect, e.g. with the DisplayInfo, and other related components on the desktop facet.

Yes, it's still relevant, but no, I haven't managed to isolate it further. It happens rarely enough that I'm not sure I'd be able to tell if changing the refresh rate would have any impact, but it does still happen (I notice it every couple of weeks, probably happens more often without me noticing). And it's frequent enough to be really annoying.

I will take a look at the components in the working condition and then next time it happens compare them to the broken state, to see if there is a difference.

**Comment by JackTheFoxOtter** [comment]
I've taken screenshots of the 3 different screens in working condition, now I'll have to wait for it to happen again, then I will see if there is anything different in broken condition.

For my own reference:
![2024-04-23 18 14 03](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/345232ca-9066-4b07-9a1b-1954046c1760)
![2024-04-23 18 14 10](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/57d49b48-4d10-4a4d-bb8c-49fc5f614179)
![2024-04-23 18 14 15](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/e4fd7b85-3775-4bd0-8f47-c00fbb525a76)

**Comment by dragos240** [comment]
I have this problem too. It's definitely still a problem. This happens to me fairly often, like half the time. I have a feeling it may happen when a window on one screen goes into "notify mode". The other screen freezes, but the one with the "active" window continues to work. It's always the same monitor, the non-default monitor. I also don't think it's an OS issue because SteamVR's desktop controls work just fine when this occurs.

**Comment by shiftyscales** [comment]
Can you clarify what you mean by "notify mode", @dragos240? Could you provide reference screenshots/video? Could you please also provide specs about your monitors as Jack did above? It would be useful to determine if there are any other common variables you share.

**Comment by dragos240** [comment]
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/704081/ec92d4d9-dc66-4bf1-8f86-d5f469d47cef)
I meant this. I'm not sure what it's called. Either way, I was wrong. This isn't causing the problem since it was happening even when I made sure nothing had that status.

Here's my monitor configuration: 
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/704081/842556da-6ecf-41f0-a83d-59bc1381867a)

Screen 2 (leftmost, this is the only one that freezes): 
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/704081/462dc2bd-57c5-4b76-8216-c20066fa2eae)

Screen 1 (this never freezes): 
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/704081/d15d95a2-f972-4b27-84df-b006f38c01a6)

**Comment by shiftyscales** [comment]
Thank you- so like Jack, you have variances in refresh rate and resolution. It would be useful to try and isolate the relevance of those variables, so if you could lower your resolution/refresh rate to match exactly across all displays- it would be useful to see if the issue persists, @JackTheFoxOtter @dragos240.

**Comment by dragos240** [comment]
I haven't got it lately. Maybe an update fixed it? Or maybe it's just less common now. I didn't change anything aside from updating GPU drivers, which... may have fixed it come to think of it

**Comment by JackTheFoxOtter** [comment]
All right, it has happened again. (I though the issue disappeared, it's been a while since I've last noticed). The frozen screen this time was the last one (index 2)

So, following are the screenshots of the components I took:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/1df6ed06-a399-434c-9b71-bc27d03ed9f4)
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/eec74e6e-2cf3-4ba7-9274-0ae2eaf4a349)
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/35171546/5aeae1ea-398e-4cf1-a29d-ffe678bf6581)

Unfortunately I don't think there's anything on those components that could help diagnose this.
I tried replacing the DesktopTextureProvider with a new one, but that didn't fix the issue. I think whatever breaks is happening internally. No amount of re-assigning references in-game managed to resolve the issue. But I did notice something new, the right desktop isn't actually completely frozen. There was a small square around the mouse cursor sometimes in the correct position, but no window was visible in that small square, only my wallpaper from what I could tell.

[J4-C - 2024.6.11.74 - 2024-06-15 16_18_23.log](https://github.com/user-attachments/files/15849153/J4-C.-.2024.6.11.74.-.2024-06-15.16_18_23.log)

That's the log file from this session. I noticed the screen being frozen ~19:05, but I can't find anything indicating an exception with the desktop view in there. I'm not quite sure how to isolate it further.

> Thank you- so like Jack, you have variances in refresh rate and resolution. It would be useful to try and isolate the relevance of those variables, so if you could lower your resolution/refresh rate to match exactly across all displays- it would be useful to see if the issue persists, @JackTheFoxOtter @dragos240.

Considering how infrequent / random this issue happens, I don't think I could really validate if changing those values makes a difference. If I can figure out how to force it to break I could see if changing my resolution / refresh rate works, but I'm not going to reduce it potentially for months without being sure if it makes a difference.

**Comment by shiftyscales** [comment]
As a sanity check, @JackTheFoxOtter - did you happen to catch what was going on with your physical monitors at the time this happened? I have noticed before that if Windows throws up a UAC prompt- it can prevent screens in Resonite from updating until it is addressed/dismissed.

Likewise- Discord has an option to hide the Discord window from screen capture when Streamer Mode is enabled- and I've noticed that can also have a similar effect.

I don't suppose you have happened to get the `Player.log` from that session, @JackTheFoxOtter? I don't immediately see anything that stands out- but I am guessing this is possibly some exception being thrown by the desktop duplication API - particularly if there's nothing visibly obviously wrong on Resonite's end in terms of the component setup.

**Comment by JackTheFoxOtter** [comment]
I do not now what more information I need to provide here. I think this still happens, but super rarely. Have yet to find out any reliable steps to reproduce.
